---
title: "1-RNA_Seq_sanity_checks"
author: "Emma"
date: "10 February 2016"
output: 
  html_document: 
    keep_md: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

**Processing RNA Seq Data and Sanity Checks**
================================================

**Load Libraries**
-------------------

```{r}
require(ggplot2)
require(data.table)
require(knitr)
require(dplyr)
require(reshape)
require(reshape2)
require(pheatmap)
```

**Import Data**
----------------

```{r}
row.merged <- read.table("C:/Users/Emma/Documents/Masters/STAT540/team_treed_rats-DNA-methylation/RNASeq_data/row.merged.txt", header=TRUE, row.names = 1)
```

The data (rnadata) is arranged in the following table, displaying the gene, rpkm value, as well as the group and replication it belongs to - that is, for example, female vehicle 1, or female z 1. 
```{r}
head(row.merged)
```

Another data set was made (rpkmlog) with the same data but with log2 rpkm values, as this may be useful for some plots.

**Taking a peek at the data**
------------------------------

Getting a summary of the data set shows that there are 88,548 genes for each of the groups; FVEH, FZEB, MVEH, MZEB. This is good as each of the replicates should have 1/3 of this - 29,516 as expected! 
```{r}
summary(row.merged)
```

Looking at the rpkm values of the genes as a whole, there are a LOT that have a value of 0. Is this bad? Shouldn't every gene have some expression?  

```{r}
test <- subset(row.merged, rpkm_value == 0)
summary(test)
```

It also needs to be checked for each group, how many have a readout of 0, as this may cause some problems / bias.

The mean RPKM values for each of the gene sets are roughly the same. Again, is this good? 

```{r}
meanrpkms <- aggregate(row.merged[, 2], list(row.merged$group.fct), mean)
meanrpkms
```

**Arranging the data into sets of each group** 
-----------------------------------------------

To take a look at the correlation between each replicate for each group (FVEH etc), subsets were first made. 

```{r}
FVEH.set <- subset(row.merged, group.fct == "FVEH")
```

And then rearranged.. 
```{r}
FVEH.set <- cast(FVEH.set, gene.no + genes ~ replication, value = "rpkm_value")
head(FVEH.set)
```


The same was done for each set - FZEB, MZEB and MVEH.

```{r echo = FALSE}
MVEH.set <- subset(row.merged, group.fct == "MVEH")
MVEH.set <- cast(MVEH.set, gene.no + genes ~ replication, value = "rpkm_value")
head(MVEH.set)
```
```{r echo = FALSE}
MZEB.set <- subset(row.merged, group.fct == "MZEB")
MZEB.set <- cast(MZEB.set, gene.no + genes ~ replication, value = "rpkm_value")
head(MZEB.set)
```
```{r echo=FALSE}
FZEB.set <- subset(row.merged, group.fct == "FZEB")
FZEB.set <- cast(FZEB.set, gene.no + genes ~ replication, value = "rpkm_value")
head(FZEB.set)
```

Now they can be correlated:

**Female Vehicle**

```{r}
cor(FVEH.set$"1", FVEH.set$"2")
cor(FVEH.set$"1", FVEH.set$"3")
cor(FVEH.set$"2", FVEH.set$"3")
```

**Male Vehicle**

```{r}
cor(MVEH.set$"1", MVEH.set$"2")
cor(MVEH.set$"1", MVEH.set$"3")
cor(MVEH.set$"2", MVEH.set$"3")
```

**Male Treated (Zeb)**

```{r}
cor(MZEB.set$"1", MZEB.set$"2")
cor(MZEB.set$"1", MZEB.set$"3")
cor(MZEB.set$"2", MZEB.set$"3")
```

**Female Treated (Zeb)**

```{r}
cor(FZEB.set$"1", FZEB.set$"2")
cor(FZEB.set$"1", FZEB.set$"3")
cor(FZEB.set$"2", FZEB.set$"3")
```

The lowest correlations are in Male Vehicle, but at 0.97.. and 0.98... I don't think that's bad! 

```{r eval=TRUE, echo=FALSE}
names(MVEH.set)[3:5] <- c("MVEH 1", "MVEH 2", "MVEH 3")
names(FVEH.set)[3:5] <- c("FVEH 1", "FVEH 2", "FVEH 3")
names(FZEB.set)[3:5] <- c("FZEB 1", "FZEB 2", "FZEB 3")
names(MZEB.set)[3:5] <- c("MZEB 1", "MZEB 2", "MZEB 3")
```

Merge it all into a nice table (allmerged): 

```{r echo=FALSE}
mergemf <- merge(MVEH.set, FVEH.set)
mergemf2 <- merge(mergemf, MZEB.set)
allmerged <- merge(mergemf2, FZEB.set)
head(allmerged)
```

```{r}
matrix <- subset(allmerged, select = 3:14)
pairwise_cor <- matrix %>%
   cor(use = "pairwise.complete.obs")
diag(pairwise_cor) <- 1
pheatmap(pairwise_cor, cluster_rows = F, cluster_cols = F, display_numbers = T)
```

As the heatmap is comparing the mean of all gene expression for each sample, we would expect the correlation between each to be high. 

