---
title: "Analysis pipeline for finding differentially expressed genes from RNAseq `rpkm` read counts using `R` package `NOISeq`"
author: "Rashed"
date: "February 24, 2016"
output: 
  html_document: 
    fig_caption: yes
    keep_md: yes
    toc: yes
---

# Data preparation for comparing male and female vehicle


```{r ch11}
#setwd("team_treed_rats-DNA-methylation/RNASeq_data/RNASeq_data")
#dir()

datn <- read.table(file=
                     "C:/Users/Rashed/Documents/all_github/team_treed_rats-DNA-methylation/RNASeq_data/RNAseq_all_merged.txt", 
                   header = TRUE)
rownames(datn) <- datn$genes
datn <- datn[,c(2:14)]
head(datn)
#str(datn)

###comparing male and female vehicle

####data preparation

###count data
#can input raw or norrmalized count. here we put rpkm values as count
countsfm <- datn[,c(1:3, 7:9)]
head(countsfm)
str(countsfm)
dim(countsfm)
```


## Import data to make `NOISeq object`

```{r ch12}
###removing rows with all zero counts

countsfmnz <- countsfm[countsfm[,1] != 0 | countsfm[,2] != 0 | countsfm[,3] != 0 | countsfm[,4] != 0 | countsfm[,5] != 0 | countsfm[,6] != 0 , ]
dim(countsfmnz)

###factor object for NOISeq
factorsfm <- data.frame(Vehicle = c("Female", "Female", "Female", "Male", "Male", "Male"), 
	VehicleRun = c("Female_1", "Female_2", "Female_3", "Male_1", "Male_2", "Male_3"), 
	Run = rep(c("R1", "R2", "R3"), times = 2))
rownames(factorsfm) <- colnames(countsfm)
factorsfm

###converting data into a NOISeq object
library(NOISeq)
require(NOISeq)
datafm <- NOISeq::readData(data = countsfmnz, factors = factorsfm) #using readData function from NOISeq
datafm
```


##check the infirnation included

```{r ch13}
#check what information is included
str(datafm)
head(assayData(datafm)$exprs) ###give the expression data with rpkm counts
head(pData(datafm))
head(featureData(datafm)@data)
```

#Checking Quality control of the data

##Some exploratory plot

```{r ch14}
#### Quality control of count data
#need not for already normalized count I think

##Generating data for exploratory plots
#can't do so due to lack of information on biotype detection, sequencing depth and expression 
#quantification, sequencing bias detection and batch effect exploration

##count distribution per sample
countsplo <- dat(datafm, factor = NULL, type="countsbio")
explo.plot(countsplo, toplot = 1, samples =NULL, plottype = "boxplot")

###not give much info as all samples are normalized

#sensitivity plot

explo.plot(countsplo, toplot = 1, samples =NULL, plottype = "barplot")
#showing number of features with low counts for each samples 

```


##Bias things

```{r ch15}
###Sequencing bias detection

##not needed as the countsa re already normalized

#length bias plot: he “lengthbias” plot describes the relationship between the feature length 
#and the expression values. For each bin, the 5% trimmed mean of the corresponding expression
#values (CPM if norm=FALSE or values provided if norm=TRUE) is computed and depicted in Y axis.

#lengthbiasfm <- dat(datafm, factor = "Vehicle", norm = TRUE, type = "lengthbias")
#explo.plot(lengthbiasfm, samples = NULL, toplot = "global")
#show(lengthbiasfm)
#can't run as we haven't the feature length

####RNA composition

###to check bias cd plot is used

fmcd <- dat(datafm, type = "cd", norm = TRUE, refColumn = 1)

explo.plot(fmcd)

###pca plot

fmPCA = dat(datafm, type = "PCA")
explo.plot(fmPCA, factor = "Vehicle")
```

## QC report

```{r ch16}
###Quality control report

QCreport(datafm, samples = NULL, factor = "Vehicle", norm = TRUE)
#diagnostic test failed.
#result shows normalization is required to correct for bias....confused!!!

```


#Normalization

```{r ch17}
###Normalization
#not needed as normalization is done already.
```


# Low-count Filtering

```{r ch18}
###Low-count filtering
#Excluding features with low counts improves, in general, differential expression results, 
#no matter the method being used, since noise in the data is reduced.

#NOISeq includes three methods to filter out features with low counts: 1. CPM 2. Wilcoxon test 
#3. Proportion test

##Using CPM
fmfilt <- filtered.data(countsfmnz, factor = factorsfm$Vehicle, norm = TRUE, 
                       depth = NULL, method = 1, cv.cutoff = 100, cpm = 1, p.adj = "fdr")
#Filtering out low count features...
#14458 features are to be kept for differential expression analysis with filtering method 1
#The “Sensitivity plot” described in previous section can help to take decisions on the CPM 
#threshold to use in methods 1 and 3. 

##Using proportion test
#cant do as we haven't sequence depth info
```

#Differential Expression

##Several possible options

```{r ch19}
####Differential expression

#NOISeq-real: using available replicates
fmnoiseq1 <- noiseq(datafm, k = 0.5, norm = "n", factor = "Vehicle", nss = 0, replicates = "technical")
head(fmnoiseq1@results[[1]])

fmnoiseq2 <- noiseq(datafm, k = 0.5, norm = "n", factor = "Vehicle", nss = 5, replicates = "biological")
head(fmnoiseq2@results[[1]])

#NOISeqBIO: recommended when we have biological replicates
fmnoiseqbio <- noiseqbio(datafm, k = 0.5, norm = "n", factor = "Vehicle", 
                         r = 20, adj = 1.5, plot = TRUE, a0per = 0.9, random.seed = 12345, 
                         filter = 1)
```


##Select the differentially expressed features/genes

Please remember that, when using `NOISeq`, the probability of differential expression is not equivalent to 1 − pvalue. We recommend for q to use values around 0.8. However, when using `NOISeqBIO`, the probability of differential expression would be equivalent to 1 − F DR, where FDR can be considered as an adjusted p-value. Hence, in this case, it would be more convenient to use q = 0.95. With the argument M we choose if we want all the differentially expressed features, only the differentially expressed features that are more expressed in condition 1 than in condition 2 (M = “up”) or only the differentially expressed features that are under-expressed
in condition 1 with regard to condition 2 (M = “down”):

```{r ch20}
fmnoiseq1.deg <- degenes(fmnoiseq1, q = 0.8, M = NULL)

fmnoiseq1.deg1 <- degenes(fmnoiseq1, q = 0.8, M = "up")

fmnoiseq1.deg2 <- degenes(fmnoiseq1, q = 0.8, M = "down")

fmnoiseq2.deg <- degenes(fmnoiseq2, q = 0.8, M = NULL)

fmnoiseq2.deg1 <- degenes(fmnoiseq2, q = 0.8, M = "up")

fmnoiseq2.deg2 <- degenes(fmnoiseq2, q = 0.8, M = "down")

fmnoiseqbio.deg <- degenes(fmnoiseqbio, q = 0.95, M = NULL)

fmnoiseqbio.deg1 <- degenes(fmnoiseqbio, q = 0.95, M = "up")

fmnoiseqbio.deg2 <- degenes(fmnoiseqbio, q = 0.95, M = "down")
```

I am confused at the output found!!! Seems too less differential genes. 



