---
title: "2-RNAseq_differential_expression"
author: "Tony Hui and Rashed"
date: "March 8, 2016"
output: 
  html_document: 
    keep_md: yes
    self_contained: yes
---

```{r setup, message=FALSE, warning=FALSE}
library(NOISeq)
library(ggplot2)
require(tidyr)
require(knitr)
require(limma)
require(edgeR)
require(gplots)
require(pheatmap)
library(dplyr)
```

```{r, eval=FALSE}
setwd("Data_Analysis/")
```

## Load data

```{r}
rnaseq <- read.table(file="../RNASeq_data/new_data_Tony_TPM/RNAseq_new_merged_raw.txt", header = TRUE, stringsAsFactors = FALSE)

rnaseq_meta <- read.table(file = "../RNASeq_data/new_data_Tony_TPM/sailfish_file_table.txt", stringsAsFactors = FALSE)

colnames(rnaseq) <- with(rnaseq_meta, paste(V3, V4, rep(1:3, 3), sep = "_"))

rnaseq_meta$samples <- with(rnaseq_meta, paste(V3, V4, rep(1:3, 3), sep = "_"))
```

## Plot distribution of gExps

```{r}
rnaseq_male_Female <- rnaseq %>%
  add_rownames("gene") %>%
  select(gene, contains("vehicle")) %>%
  gather(key = sample, value = gExp, -gene) %>%
  mutate(gender = ifelse(grepl("Female", sample), "Female", "male"))

rnaseq_male_Female %>% 
  ggplot(aes(gExp+0.5, color = gender)) +
  geom_density() +
  scale_x_log10()
```

## Try Limma's `voom` function

```{r}
samples <- rnaseq_meta %>% filter(V4 == "vehicle")

limma_design_matrix <- model.matrix(~V3, samples)

rownames(limma_design_matrix) <- samples$samples

voom_DGElist <- rnaseq %>%
  select(contains("vehicle")) %>%
  DGEList(group = rep(c("f","m"), each = 3)) %>%
  .[rowSums(cpm(.) > 0.3) >= 2, , keep.lib.sizes=FALSE]

voom_rnaseq <- voom_DGElist %>%
  voom(design = limma_design_matrix, plot = T)

fit_limma <- lmFit(object = voom_rnaseq, design = limma_design_matrix) %>% eBayes()

limma_results <- topTable(fit_limma, adjust="fdr", number = Inf)
```

Pvalues are skewed to the right

```{r}
limma_results %>% 
  ggplot(aes(P.Value)) +
  geom_density()
```

### Double-check nothing funny is going on

```{r}
correlation <- cor(rnaseq %>% select(contains("vehicle")), method = "spearman")

diag(correlation) <- NA

clustering <- hclust(as.dist(1-correlation), method = "ward.D2")

require(pheatmap)
pheatmap(correlation, cluster_rows = clustering, cluster_cols = clustering, display_numbers = T, color = colorRampPalette(c("#ffffb2", "#bd0026"))(9))

plot(clustering)
```

Seems like everything is normal, although there doesn't seem to be a clear separation between male and female

## Try edgeR with usual Negative Binomial fit

```{r}
edgeR_DGElist <- rnaseq %>%
  select(contains("vehicle")) %>%
  DGEList(group = rep(c("f","m"), each = 3)) %>%
  .[rowSums(cpm(.) > 0.3) >= 2, , keep.lib.sizes=FALSE] %>%
  calcNormFactors()

edgeR_DGElist_trends <- edgeR_DGElist %>%
  estimateGLMCommonDisp(limma_design_matrix, verbose=TRUE) %>%
  estimateGLMTrendedDisp(limma_design_matrix) %>%
  estimateGLMTagwiseDisp(limma_design_matrix)

plotBCV(edgeR_DGElist_trends)

# plotMDS.DGEList(edgeR_DGElist_trends)
```

```{r}
fit <- glmFit(edgeR_DGElist_trends, limma_design_matrix) %>% glmLRT(coef = 2)

edgeR_results <- topTags(fit, n = Inf) %>% as.data.frame()

edgeR_results %>% head() %>% kable("markdown")
```

Once again, right-skewed Pvalues

```{r}
qplot(edgeR_results$PValue, geom="density")
```

## Try edgeR with Negative Binomial quasi-likelihood(QL) fit

There is `glmFit` for the previous one and `glmQLFit` for this one which incorporates quasi-likelihood fitting. The main differences between these two are as follows:

`glmQLFit` actually addresses two type of dispersions: one is the gene-specific dispersion modeled by QL dispersion parameter and another one is the NB dispersion parameter, the global one over all genes.

"So the two dispersion parameters have different roles. The NB dispersion describes the overall biological variability across all genes. The square-root of the NB dispersion is known as the biological coefficient of variation. It represents the observed variation that is attributable to inherent variability in the biological system, in contrast to the Poisson variation from sequencing. The QL dispersion picks up any gene-specific variability above and below the overall level. " (Quoted from the reference below.) Both are incorporated in the glmQLFit. 

For full reference with more discussion you can find the reference here: [Link](http://www.statsci.org/smyth/pubs/QLedgeRPreprint.pdf) 

```{r}
fitQL <- glmQLFit(edgeR_DGElist_trends, limma_design_matrix) %>% glmLRT(coef = 2)

edgeR_QL_results <- topTags(fitQL, n = Inf) %>% as.data.frame()

edgeR_QL_results %>% head() %>% kable("markdown")
```

Once again, right-skewed Pvalues

```{r}
qplot(edgeR_QL_results$PValue, geom="density")
```

## Summary of results

```{r, echo=FALSE}
paste("there are", nrow(limma_results %>% filter(adj.P.Val<0.05)), "DE genes from limma")
paste("there are", nrow(edgeR_QL_results %>% filter(FDR<0.05)), "DE genes from edgeR glmQLFit")
paste("there are", nrow(edgeR_results %>% filter(FDR<0.05)), "DE genes from edgeR glmFit")
```
```{r}
venn(list(
  edgeR_QL = edgeR_QL_results %>% subset(FDR<0.05) %>% rownames(.),
  limma = limma_results %>% subset(adj.P.Val<0.05) %>% rownames(.),
  edgeR = edgeR_results %>% subset(FDR<0.05) %>% rownames(.)
))
```

## Check for cannonical gene that should be differentially expressed

```{r}
rn6_gene <- read.table("../Data_Analysis/rn6_genes.txt") %>% tbl_df() %>%
  select(gene = V1, V7) %>% 
  unique()

cannonical_gene <- c("Prl", "Xist", "Dby", "Eif2s3y", "Rps4y2", "Smcy", "Uty", "Eif2s3")

rn6_gene_interest <- rn6_gene %>%
  filter(V7 %in% cannonical_gene)

rn6_gene_interest
```

```{r}
right_join(rn6_gene, edgeR_QL_results %>% add_rownames("gene"), by = "gene")  %>%
  filter(gene %in% rn6_gene_interest$gene) %>% kable("markdown")
```

Looks like some of cannonical genes are differentially expressed. Yay!

```{r}
# write.table(edgeR_QL_results %>% subset(FDR<0.05) %>% rownames(.), file = "/projects/epigenomics/users/thui/stat540/methylation_data/homer/de_transcripts.txt", row.names = F, col.names = F, quote = F)
```
```{r}
edge_QL_final <- edgeR_QL_results %>% subset(FDR<0.05) %>% add_rownames("gene")

edge_QL_final <- edge_QL_final %>%
  select(gene, FDR)

gExp <- 
  rnaseq %>%
  # select(contains("vehicle")) %>%
  DGEList() %>%
  calcNormFactors() %>%
  cpm() %>% 
  as.data.frame() %>%
  add_rownames("gene") %>%
  # filter(gene %in% edge_QL_final$gene) %>%
  inner_join(., edge_QL_final) %>%
  gather(key = sample, value = gExp, -gene) %>%
  mutate(group = gsub("\\_[1-9]", "", sample)) %>%
  group_by(gene, group) %>%
  summarize(mean = mean(gExp) %>% round(3)) %>%
  spread(key = group, value = mean) %>%
  ungroup() 

# write.table(gExp, file = "../Data_Analysis/RNAseq_result/DE_genes/maleVSfemale_glmQLFit_DE_genes.tsv", row.names = F, col.names = T, quote = F, sep = "\t")
```

## Compare vs NOIse-seq

```{r, warning=FALSE}
rnaseq_samples <- rnaseq %>%select(contains("vehicle"))

noiseq_factors <- data.frame(gender = rep(c("female", "male"), each=3), row.names = colnames(rnaseq_samples))

noiseq_data <- readData(data = rnaseq_samples, factors = noiseq_factors)

noiseq_results <- noiseqbio(input = noiseq_data, factor = "gender", norm = "tmm", filter = 1)

noiseq_hits <- degenes(noiseq_results)

DE.plot(output = noiseq_results, q = 0.95, graphic = "expr")
```
```{r}
x<-venn(list(
  edgeR_glmQLFit = edgeR_QL_results %>% subset(FDR<0.05) %>% rownames(.),
  noiseSeq = rownames(noiseq_hits),
  edgeR_glmFit = edgeR_results %>% subset(FDR<0.05) %>% rownames(.)
))
x <- attr(x, "intersection")
```

## Noticed that `r length(x$noiseSeq)` genes are only called DE in `NOISeq`

Let's take a look at these genes

```{r, eval=TRUE}
full_data <- rnaseq_samples %>%
  DGEList(group = rep(c("f","m"), each = 3))

cpm(full_data) %>% subset(rownames(.) %in% x$noiseSeq) %>% round(2) %>% kable("markdown")

cpm(full_data) %>% subset(rownames(.) %in% x$noiseSeq) %>% 
  pheatmap(scale = "row")
```

Basically all of them are only expressed in one sample, which may be a technical artifact. AKA these genes are no good, and shouldn't be considered. 

## Grand conclusion

Proceed with `glmQLFit` since it gives the most number of genes, presumably due to highest sensitivity