---
title: "3 - DE_genes_femaleVSfemaleZeb"
author: "Tony"
date: "3/23/2016"
output: 
  html_document: 
    keep_md: yes
---

```{r setup, message=FALSE, warning=FALSE}
library(dplyr)
library(ggplot2)
require(tidyr)
require(knitr)
require(edgeR)
require(gplots)
require(pheatmap)
```

## Load data

```{r}
femaleVSmale_de_genes <- read.table("../Data_Analysis/RNAseq_result/DE_genes/glmQLFit_DE_genes.tsv", header = TRUE)

rnaseq <- read.table(file="../RNASeq_data/new_data_Tony_TPM/RNAseq_new_merged_raw.txt", header = TRUE, stringsAsFactors = FALSE) 

dim(rnaseq)

rnaseq_meta <- read.table(file = "../RNASeq_data/new_data_Tony_TPM/sailfish_file_table.txt", stringsAsFactors = FALSE)

colnames(rnaseq) <- with(rnaseq_meta, paste(V3, V4, 1:12, sep = "_"))

rnaseq_meta$samples <- with(rnaseq_meta, paste(V3, V4, 1:12, sep = "_"))

rn6_gene <- read.table("rn6_genes.txt") %>% tbl_df() %>%
  select(gene = V1, V7) %>% 
  unique()
```

## Prepare DGEList object and design matrix

```{r}
rnaseq_subset <- rnaseq %>% 
  # subset genes to only the genes that were differentially expressed between male and female
  subset(rownames(.) %in% femaleVSmale_de_genes$gene)

edgeR_DGElist_females <- rnaseq_subset %>%
  select(contains("female")) %>%
  DGEList(group = rep(c("f","fz"), each = 3)) %>%
  calcNormFactors(method = "TMM") 

design_matrix <- rnaseq_meta %>% filter(V3 == "Female") %>% model.matrix(~V4, .)
rownames(design_matrix) <- edgeR_DGElist_females$samples %>% rownames()

edgeR_DGElist_females_trends <- edgeR_DGElist_females %>%
  estimateGLMCommonDisp(design_matrix, verbose=TRUE) %>%
  estimateGLMTrendedDisp(design_matrix) %>%
  estimateGLMTagwiseDisp(design_matrix)

plotBCV(edgeR_DGElist_females_trends)
plotMDS.DGEList(edgeR_DGElist_females_trends)
```

## Check for outliers

```{r}
correlation <- cor(rnaseq_subset %>% select(contains("female")), method = "spearman")

diag(correlation) <- NA

clustering <- hclust(as.dist(1-correlation), method = "ward.D2")

require(pheatmap)
pheatmap(correlation, cluster_rows = clustering, cluster_cols = clustering, display_numbers = T, color = colorRampPalette(c("#ffffb2", "#bd0026"))(9))

plot(clustering)
```

Female Vehicle and Female Zeb separate out nicely (based on spearman correlation)

## Call DEGs

```{r}
fitQL <- glmQLFit(edgeR_DGElist_females_trends, design_matrix) %>% glmLRT(coef = 2)

edgeR_QL_results <- topTags(fitQL, n = Inf) %>% as.data.frame()

edgeR_QL_results %>% head() %>% kable("markdown")
```

```{r}
qplot(edgeR_QL_results$PValue)
```

```{r, echo=F}
paste("there are", nrow(filter(edgeR_QL_results, FDR < 0.05)), "DE genes at FDR < 0.05")
```

Very little DE genes - let's see what's happening

```{r}
qplot(edgeR_QL_results$logFC, edgeR_QL_results$FDR) + scale_y_log10()
```

Zoom in... and let's see where the FDR = 0.2 line is

```{r}
qplot(edgeR_QL_results$logFC, edgeR_QL_results$FDR) + scale_y_log10(limits = c(1e-2, 1)) + geom_hline(yintercept = 0.2)
```

```{r, echo=F}
paste("there are", nrow(filter(edgeR_QL_results, FDR < 0.2)), "DE genes at FDR < 0.2")
```

Seems more legit to use fdr < 0.2 in this case

```{r}
femVsfemZeb_edge_QL_final <- edgeR_QL_results %>% subset(FDR<0.2) %>% round(3) %>% add_rownames("gene")

femVsfemZeb_edge_QL_final %>% head %>% kable("markdown")
```

## Filter genes that are both either up or downregulated compared to female

```{r}
output_results <- rnaseq_subset %>% 
  select(-starts_with("Male_zeb")) %>%
  add_rownames("gene") %>%
  filter(gene %in% femVsfemZeb_edge_QL_final$gene) %>%
  gather(key = group, value = gExp, -gene) %>%
  mutate(group = as.character(gsub("\\_[1-9]", "", group))) %>%
  # head(10) %>%
  group_by(gene, group) %>%
  summarize(mean_exp = mean(gExp) %>% round(digits = 2)) %>%
  spread(key = group, value = mean_exp)
  
output_results_TF <- output_results %>%
  mutate(gExp_up_in_femaleVSmale = Female_vehicle > Male_vehicle,
         gExp_up_in_femaleVSzeb = Female_vehicle > Female_zeb,
         concordant = gExp_up_in_femaleVSmale == gExp_up_in_femaleVSzeb)

output_results_TF %>% head() %>% kable("markdown")
```

Call genes concordant if their expression in Fem+Zeb and Male changes in the same way compared to Female *(eg both upregulated compared to female)*

```{r}
table(output_results_TF$concordant)
```

Good, most genes are concordant

### Just a quick heatmap based on these genes

```{r}
gene_list <- output_results_TF %>% filter(concordant) %>% .$gene
plot_heatmap <- subset(rnaseq, rownames(rnaseq) %in% gene_list) %>% select(-starts_with("male_zeb"))
plot_heatmap <- log(plot_heatmap+1, base=10)
```

Color scale represents `log10(count+1)`

```{r, fig.width=4, fig.height=6, dpi=150}
pheatmap(plot_heatmap, show_rownames = F, cluster_cols = hclust(as.dist(1-cor(plot_heatmap, method = "spearman")), method = "ward.D2"), clustering_method = "ward.D2")
```

Try z-score normalization within rows

```{r, fig.width=4, fig.height=6, dpi=150}
pheatmap(plot_heatmap, show_rownames = F, scale = "row", cluster_cols = hclust(as.dist(1-cor(plot_heatmap, method = "spearman")), method = "ward.D2"), clustering_method = "ward.D2")
```

## save results

Print only the concordant genes

```{r, warning=FALSE}
output_results_final <- output_results_TF %>%
  filter(concordant) %>%
  left_join(., femVsfemZeb_edge_QL_final) %>%
  select(-concordant) %>%
  inner_join(., rn6_gene) %>%
  select(-LR, -logCPM)

output_results_final %>%
  write.table(file = "../Data_Analysis/RNAseq_result/DE_genes/femVSfemZeb_glmQLFit_DE_genes.tsv", row.names = F, col.names = T, quote = F, sep = "\t")
```
