Emma Laks

**References for poster**
=========================

[1] Nugent et al. Brain feminization requires active repression of masculinization via DNA methylation, Nature Neuroscience (2015) 18:690–697. [doi:10.1038/nbt.2862](http://www.nature.com/neuro/journal/v18/n5/full/nn.3988.html)

[2] Heinz et al. Simple Combinations of Lineage-Determining Transcription Factors Prime cis-Regulatory Elements Required for Macrophage and B Cell Identities. Mol Cell (2010) 28;38(4):576-589. 

[3] Patro et al. Sailfish enables alignment-free isoform quantification from RNA-seq reads using lightweight algorithms. Nature Biotechnology (2014) 32:462–464. [doi:10.1038/nbt.2862](http://www.nature.com/nbt/journal/v32/n5/full/nbt.2862.html)

[ref1] van Arensbergen et al. In search of the determinants of enhancer–promoter interaction specificity. Trends in Cell Biology (2014) 24(11):695–702. [doi:10.1016/j.tcb.2014.07.004](http://www.cell.com/trends/cell-biology/fulltext/S0962-8924(14)00118-4)

[ref2] Ziller et al. Coverage recommendations for methylation analysis by whole-genome bisulfite sequencing. Nature Methods (2015) 12:230–232. [doi:10.1038/nmeth.3152](http://www.nature.com/nmeth/journal/v12/n3/full/nmeth.3152.html)

[CITE1] Sanabra et al. Neuroanatomical distribution and neurochemical characterization of cells expressing adenylyl cyclase isoforms in mouse and rat brain. Journal of Chemical Neuroanatomy (2011) 41(1):43–54. [doi:10.1016/j.jchemneu.2010.11.001](http://www.sciencedirect.com/science/article/pii/S0891061810001535)

[CITE2] Visel et al. Comprehensive analysis of the expression patterns of the adenylate cyclase gene family in the developing and adult mouse brain. Journal of Comparative Neurology (2006) 496(5):684–697. [doi:10.1002/cne.20953](http://onlinelibrary.wiley.com/doi/10.1002/cne.20953/full)

[CITE3] M. Uittenbogaard et al. Expression of the bHLH transcription factor Tcf12 (ME1) gene is linked to the expansion of precursor cell populations during neurogenesis. Gene Expression Patterns (2002) 1(2):115–121. [doi:10.1016/S1567-133X(01)00022-9](http://www.sciencedirect.com/science/article/pii/S1567133X01000229)


**Epigenetic Determinants of Gender in Rats - background research**
===================================================================
*Summary*

 + Did literature review on rats, mice, and humans with a focus on brain tissue, and differential gene expression between males and females
 + Based on this, created a big file from various supplements of DEGs in male and female brain tissue
 + Checked this against our DEGs
 
 Genes having concordance between literature and our m/f DEGs include:
  + Eif2s3
  + Mef2c
  + Sesn1
  + Esr1



**Notes for background research**
===================================================================
[A rat RNA-Seq transcriptomic BodyMap across 11 organs and 4 developmental stages - 2013](http://www.nature.com.ezproxy.library.ubc.ca/ncomms/2014/140210/ncomms4230/full/ncomms4230.html)
- has RNA-seq data for rats on differentially expressed genes (DEGs) by sex in 9 non-sex organs at 4 dev. stages. Could be useful for comparison data... Relevant text, emphasis mine:

_"Even though they represented a **small proportion of overall variation in gene expression**, differential gene expression profiles between female and male rats for all nine non-sex organs were examined at all four developmental stages (Figs 4a,b, and Supplementary Fig. 10). A number of genes were significantly different between male and female rats, particularly in the liver, muscle and kidney, and to a lesser extent in the spleen and brain. **Most DEGs were found at 21 weeks,** in adults (Supplementary Table 5). **More notable were, at 6 weeks, the 2,230 female-dominant genes (sexually dimorphic expression with higher expression in female) compared with 1,668 male-dominant genes (sexually dimorphic expression with higher expression in male). Female-dominant genes were outnumbered by male-dominant genes at all other ages** (female versus male: 1,921 versus 3,409 for week 2; 2,769 versus 2,945 for week 21; and 2,116 versus 2,571 for week 104). More genes showed sex-specific expression in the liver and kidney in week 21 with large FCs (Fig. 4a,b, and Supplementary Table 5). Genes involved in metabolism, particularly cytochrome P450s, are also known to be differentially expressed between the sexes. We found P450 differences to be variable; however, expression levels of Cyp1a1, Cyp1a2, Cyp2c7, Cyp3a9 and Cyp26a1 were higher in the female liver, predominantly at sexual maturity, whereas Cyp2a2, Cyp2c, Cyp3a2 and Cyp3a18 were expressed higher in the male liver (data not shown). **Major known functions of the 6,677 sex-specific DEGs annotated in RefSeq included cell cycle, blood coagulation and CREM signalling in the testis and GABA-B receptor signalling in presynaptic nerve terminals** (Fig. 2d and Supplementary Data 4).
Using a gene with many different alternatively spliced variants as an illustration, we also explored the organ-dependent and sex-specific differential isoform expression of Ugt1a1 (UDP glucuronosyltransferase 1 family, polypeptide A1), an enzyme playing an essential role in the detoxification of xenobiotics and endogenous compounds by conjugating bilirubin with glucuronic acid28, 29, 30. Twelve Ugt1a1 isoforms were annotated for rat in AceView, two of which (Ugt1a1 g and h) showed organ-dependent differential expression between female and male rats. Ugt1a1 h was expressed significantly higher in female liver, while Ugt1a1 g was more highly expressed in the male adrenal gland and lung (Supplementary Fig. 11). The Ugt1a1 gene itself, as well as its other 10 isoforms (data not shown), did not show any sex-specific differential expression."_

Also note: _"We observed that the **sex difference was subtle and only accounted for 0.22% of overall variance in expression profiles**. It should be noted that t**he Y chromosome of the rat has not been sequenced and annotated**, explaining the relatively small between-sex differences observed in our data."_

Note Supplementary [Figure S10](http://www.nature.com.ezproxy.library.ubc.ca/ncomms/2014/140210/ncomms4230/extref/ncomms4230-s1.pdf). for their scheme for identifying sex-specific genes in each organ at each developmental stage."Fold change and t-test p-value were calculated between males and females in each organ across the four stages. Sex-specific genes were those that had a fold change ≥ 2 (or ≤0.5) and p-value ≤ 0.05 for any organ at any development stage."

Also note Supplementary Data 4.  GeneGo canonical significantly **enriched pathway maps based on sex-specific genes.**

----

[Epigenetic Underpinnings of Developmental Sex Differences in the Brain - 2011] (http://www.karger.com.ezproxy.library.ubc.ca/Article/Fulltext/325264)

- Review paper, focus isn't all rat, but has links to papers that do. _"The data outlined in this review provide evidence that gonadal hormones in the neonatal brain influence epigenetic processes such as DNA methylation and histone acetylation, but also **emphasize the primitive status of our current understanding of epigenetics and sexual differentiation and the brain**."_

- Worth looking for ERα promoter methylation status if possible:
_"**In the rat, five alternative promoters control ERα gene expression. A system of untranslated exons and regulatory elements determine ERα transcription initiation** [56,57,58]. Estrogens are known to regulate the selective usage of these promoter systems [59] and **although it has not been explicitly tested, sex differences in ERα promoter methylation may be related to alternative promoter selection and thereby involved in sexual differentiation of the brain.** The [preoptic area] POA is a highly sexually dimorphic region necessary for male sex behavior [60,61]. An analysis of ERα alternative promoter usage in the POA following exposure to endocrine disrupting chemicals revealed that sex and exposure to estrogenic compounds determines the methylation status and usage of alternative promoters. These differences are dependent on both developmental timepoint and compound dosage [62]."_

REFs
[56](http://www.ncbi.nlm.nih.gov/pubmed/3031601): Koike S, Sakai M, Muramatsu M: Molecular cloning and characterization of rat estrogen receptor cDNA. Nucleic Acids Res 1987;6:2499–2513.

[57](http://www.ncbi.nlm.nih.gov/pubmed/8830677): Hirata S, Koh T, Yamada-Mouri N, Hoshi K, Kato J: The untranslated first exon ‘exon OS’ of the rat estrogen receptor gene. FEBS Lett 1996;3:371–373.

[58](http://www.ncbi.nlm.nih.gov/pubmed/12856802): Hirata S, Koh T, Yamada-Mouri N, Kato J: The novel untranslated first exon ‘exon 0N’ of the rat estrogen receptor gene. Biochem Biophys Res Commun 1996;2:653.

[59](http://www.ncbi.nlm.nih.gov/pubmed/10551786): Donaghue C, Westley B, May F: Selective promoter usage of the human estrogen receptor-α gene and its regulation by estrogen. Mol Endocrinol 1999;11:1934–1950.

[60](http://www.ncbi.nlm.nih.gov/pubmed/2611691): De Jonge F, Louwerse A, Ooms M, Evers P, Endert E, Van de Poll N: Lesions of the SDN-POA inhibit sexual behavior of male Wistar rats. Brain Res Bull 1989;6:483–492.

[61](http://www.ncbi.nlm.nih.gov/pubmed/656937): Gorski RA, Gordon J, Shryne J, Southam A: Evidence for a morphological sex difference within the medial preoptic area of the rat brain. Brain Res 1978;2:333–346.

[62](http://www.ncbi.nlm.nih.gov/pubmed/17592034): Monje L, Varayoud J, Luque EH, Ramos JG: Neonatal exposure to bisphenol A modifies the abundance of estrogen receptor-α transcripts with alternative 5′-untranslated regions in the female rat preoptic area. J Endocrinol 2007;1:201–212.

---

[Effect of DNA Methyltransferase Activity on Sexual Differentiation of the Rodent Brain - 2012] (http://archive.hshsl.umaryland.edu/handle/10713/2150)

- This is actually **the thesis by the same first author, about the same research as the paper we selected**
- Will have more detailed information about background and experiments than Nature can, so very worth checking out

notes:
- "In rat, sexual differentiation of the brain occurs during a perinatal critical period between embryonic day 18 (E18) [beginning of testosterone production by the testes in genetic males] and approximately postnatal day 10 (PN10) [after which the female brain can no longer be masculinized by exogenous steroid treatment] 
- "in males, high levels of testosterone synthesized by the testis reach the brain where a portion is converted by the enzyme p450 aromatase to estradiol, which acts on its nuclear receptors, estrogen receptors α (ERα) and β (ERβ) to initiate defeminization and masculinization
- "when estradiol is bound both ER isoforms can act as transcription factors, thereby influencing gene expression
- "Administering testosterone or estradiol to a genetic female during the critical period masculinizes behavior in adulthood if treated with testosterone, while blocking aromatization of testosterone to estradiol or antagonizing ER binding during the critical period impedes organization of the male brain. Outside of the critical period, these effects are not seen
- **Hence, the critical period has long lasting, and _poorly understood_ effects on male and female behaviours, nervous systems**
- **"If we accept that new neurons are continuously being born in the mature brain and that synapses are continuously being broken and formed anew, in order for the pattern of responses established early in life to be maintained there must be some sort of cellular memory"**
- "The research described herein describes a potential mechanistic link between early hormone exposure and regulation of gene expression." i.e. epigenetic regulation

- goes into details on difference in brain structure
- relevant discussion of epigenetics if anyone is interested/not familiar
- methods and analysis
- some useful figures/diagrams (ex fig24)

Also note:
"Despite being the most sexually dimorphic region of the brain, there were relatively few genes with overall sex differences in expression within the POA."
"Surprisingly, 387 genes that normally do not exhibit sex differences in expression became differentially expressed when both sexes were subject to DNMT inhibition (Table 15), suggesting that some underlying sex differences in gene expression are actively suppressed by DNA methylation, resulting in a convergence as opposed to divergence in males and females" - still think this is worth considering...


Abstract fragment:
-"No exogenous manipulations have ever effectively masculinized the female POA if administered outside of the critical period for sexual differentiation, however we have shown that inhibiting DNMT activity following the close of the critical period masculinizes both neuronal morphology and behavior in the female rat. These observations prompted us to rethink our definition of feminization as a passive, "default" process, and led to the hypothesize that feminization of the brain requires active, life-long suppression of masculinization via epigenetic mechanisms. In addition, we hypothesized that sex differences in DNMT activity organize sex-specific gene expression in the neonatal POA. To identify novel candidate genes involved in methylation-dependent and -independent masculinization and feminization, we quantified differences in gene expression in the POA of male and female rat pups treated with a DNMT inhibitor or vehicle using RNA-Seq. We identified novel gene and isoform variants modulated by methylation that may underlie the divergent reproductive behaviors of males versus females.”

----

[Tissue-specific expression and regulation of sexually dimorphic genes in mice - Genome Res. - 2006] (http://genome.cshlp.org/content/16/8/995.full)
- microarray gene expression between sexes in **mice**, note only [100s of genes with sexual dimorphism in the brain] (http://genome.cshlp.org/content/16/8/995/T1.expansion.html), despite thousands in other parts of the body. Seems... odd. **Wondering if the brain tissue is being epigenetically regulated to be _more_ similar than in other tissues.** However, they note "since brain is a highly heterogeneous tissue and striking physiological differences occur between sexes in specific brain regions (Arnold 2004), **sex differences in gene expression within individual regions of the brain may well be masked when whole brain is studied**. Because of this, and the limited sensitivity of microarrays to detect low levels of gene expression, it is likely that we are underestimating the number of sexually dimorphic genes in the brain."
- looks at transcription factor binding sites!
- generally seemed genetic regulation of the tissue-specific sexual dimorphism are tissue-specific

Abstract:
_"We report a comprehensive analysis of gene expression differences between sexes in multiple somatic tissues of 334 mice derived from an intercross between inbred mouse strains C57BL/6J and C3H/HeJ. The analysis of a large number of individuals provided the power to detect relatively small differences in expression between sexes, and the use of an intercross allowed analysis of the genetic control of sexually dimorphic gene expression. Microarray analysis of 23,574 transcripts revealed that the extent of sexual dimorphism in gene expression was much greater than previously recognized. **Thus, thousands of genes showed sexual dimorphism in liver, adipose, and muscle, and hundreds of genes were sexually dimorphic in brain.** These genes exhibited highly tissue-specific patterns of expression and were enriched for distinct pathways represented in the Gene Ontology database. They also showed evidence of chromosomal enrichment, not only on the sex chromosomes, but also on several autosomes. Genetic analyses provided evidence of the global regulation of subsets of the sexually dimorphic genes, as the transcript levels of a large number of these genes were controlled by several expression quantitative trait loci (eQTL) hotspots that exhibited tissue-specific control. Moreover, many tissue-specific transcription factor binding sites were found to be enriched in the sexually dimorphic genes."_

Notes:
_"In the brain, the only functional category enriched for the entire set of sexually dimorphic genes was RNA helicase activity, such as **Ddx3x/Dbx (female-biased) and Ddx3y/Dby (male-biased)**. Defense response genes encoding chemokine ligands, heat shock proteins, and histocompatibility proteins were overrepresented in the male-biased genes. For female-biased genes, **JAK-STAT **cascade and lactation pathways were enriched owing to the presence of a single gene, prolactin (Prl). **Prl** has been previously shown to be expressed more highly in females in rat brain using in vivo radioimmunoassays (DeVito 1988) and in cultures of embryonic hypothalamic cells (Beyer et al. 1992). It encodes a hormone expressed in the pituitary gland and hypothalamus, and has been suggested to be a neuropeptide that modulates reproductive behavior such as the onset of maternal behavior (Mann and Bridges 2001). In addition to several genes previously identified in brain substructures, we observed** ∼600 novel sexually dimorphic genes in brain. These sex differences could help explain sex differences in the incidence or progression of neurological and psychiatric diseases**."_

_"Tissue-specific enrichment of transcription factor binding sites (TFBS)
As an approach to understanding the regulation of sexually dimorphic gene expression, we searched for known TFBS in the conserved sequences within 5000-bp upstream and downstream regions of the sexually dimorphic genes. Using the oPPOSUM program, which contains ∼100 verified TFBS in its database, the enrichment for TFBS was analyzed, and a tissue-specific pattern in TFBS enrichment was observed. **When all sexually dimorphic genes in individual tissues were considered, only brain genes showed significant enrichment of TFBS for four transcription factors, namely, MEF2, Pax6, NF-Y, and SAP-1** (P < 0.05, Fisher’s exact test). Among them, **NF-Y has been associated with estrogen receptors** (Wang et al. 1999; Farsetti et al. 2001). At a fold-change level >1.2, several overrepresented TFBS were identified in adipose, liver, and muscle (Supplemental Tables 13–15). Among these, only TFBS for five transcription factors (SRY, PBF, MNB1A, Dof2, and Nkx) were shared in these three tissues. It is interesting to note that the binding sites for SRY and Nkx were among these five TFBS. Nkx factors are known to be regulated by androgens and are responsible for hypothalamic morphogenesis and the neuroendocrine process of sexual development (Bieberich et al. 1996; Lee et al. 2001). **There was no overlap in the enriched TFBS between brain and any of the other tissues tested.** It is important to **note that only the analyses for adipose and muscle TFBS demonstrated FDR < 5% at the P < 0.05 level; therefore, the findings and interpretation on liver and brain TFBS have to be taken with caution.**"_

----

[Sex differences in rat brain oestrogen and progestin receptors - Nature 1982](http://www.nature.com.ezproxy.library.ubc.ca/nature/journal/v300/n5893/abs/300648a0.html)

- Continuing gathering that heterogeneity of brain tissue may matter a lot, could be our paper has a sampling issue (homogenizing whole brains as mentioned earlier), which makes life harder 

"_The brains of both male and female rats possess specific cytoplasmic receptors for oestrogens and progestins1, and it has been suggested that sex differences in the neuroendocrine and behavioural responses to these hormones result from differences in the number of steroid receptors in specific locations in the brain2. **Although there have been numerous comparisons of steroid receptors in male and female brains3–9, the question of whether there are sex differences in these receptor systems remains unresolved, due in part to the inability of available biochemical techniques to achieve sufficiently high anatomical resolution.** We have overcome this problem by applying the Palkovits punch-out technique10 to measure steroid receptors in hypothalamic nuclei known to be involved in behavioural and neuroendocrine control. We report here that castrated male rats have fewer oestrogen and progestin receptors in certain hypothalamic nuclei than females. This suggests that the relative insensitivity of male rats to feminine actions of gonadal steroids might be due in part to a lack of receptors for these steroids in highly localized brain regions._"

----

[Sexually dimorphic gene expression in the hypothalamus, pituitary gland, and cortex - Genomics 2005
](http://www.sciencedirect.com.ezproxy.library.ubc.ca/science/article/pii/S0888754305000595)
- mouse transcriptome data on different brain tissues, list notes pathways

"_We examined sex differences in the transcriptomes of hypothalamus, pituitary gland, and cortex of male and female mice using serial analysis of gene expression. In total 940,669 tags were sequenced. In **hypothalamus, 3 transcripts are differentially expressed by gender**, including growth hormone (neuromodulation) and 3β-hydroxysteroid dehydrogenase-1 (steroidogenesis). In pituitary gland, **43 transcripts are differentially expressed**, including RAS guanyl-releasing protein 2 (cell signaling), ornithine transporter (mitochondrial transport), H3 histone family 3B (chromatin structure), heterogeneous nuclear ribonucleoprotein U (chromatin remodeling), NADH dehydrogenase (mitochondrial oxidative phosphorylation), neuronatin (cell differentiation), and ribosomal protein S27a (protein metabolism). **EST X (inactive)-specific transcript antisense is expressed at a higher level in the three female organs, whereas growth hormone and NADH dehydrogenase are expressed at higher levels in female cortex.** Thus, the current study has characterized key sexual dimorphisms in the transcriptomes of the hypothalamus, pituitary, and cortex._"

----

[Widespread sex differences in gene expression and splicing in the adult human brain- Nature Comm. 2013](http://www.nature.com/ncomms/2013/131122/ncomms3771/full/ncomms3771.html)
- in adult human brain (various regions), transcriptome profiling was performed using the Affymetrix Human Exon 1.0 ST Array, 1.4 million probe sets assaying expression across each individual exon

**Especially note: Supplementary Data 3 (20 KB)
List of genes with evidence of sex-biased expression in at least one CNS region
only 53 DEGs!**


"_There is strong evidence to show that men and women differ in terms of neurodevelopment, neurochemistry and susceptibility to neurodegenerative and neuropsychiatric disease. The molecular basis of these differences remains unclear. Progress in this field has been hampered by the lack of genome-wide information on sex differences in gene expression and in particular splicing in the human brain. Here we address this issue by using post-mortem adult human brain and spinal cord samples originating from 137 neuropathologically confirmed control individuals to study whole-genome gene expression and splicing in 12 CNS regions. We show that sex differences in gene expression and splicing are widespread in adult human brain, being detectable in all major brain regions and involving 2.5% of all expressed genes. We give examples of genes where sex-biased expression is both disease-relevant and likely to have functional consequences, and provide evidence suggesting that sex biases in expression may reflect sex-biased gene regulatory structures._"

----

[Sexually dimorphic gene expression in mouse brain precedes gonadal differentiation - 2003](http://www.sciencedirect.com.ezproxy.library.ubc.ca/science/article/pii/S0169328X03003395)
- whole mouse brains at 10.5 days post coitum, microarray expression data

"_The classic view of brain sexual differentiation and behavior is that gonadal steroid hormones act directly to promote sex differences in neural and behavioral development. In particular, the actions of testosterone and its metabolites induce a masculine pattern of brain development, while inhibiting feminine neural and behavioral patterns of differentiation. However, **recent evidence indicates that gonadal hormones may not solely be responsible for sex differences in brain development and behavior between males and females.** Here we examine an alternative hypothesis that genes, by directly inducing sexually dimorphic patterns of neural development, can influence the sexual differences between male and female brains. Using **microarrays and RT-PCR, we have detected over 50 candidate genes for differential sex expression, and confirmed at least seven murine genes which show differential expression between the developing brains of male and female mice at stage 10.5 days post coitum (dpc), before any gonadal hormone influence.** The identification of genes differentially expressed between male and female brains prior to gonadal formation suggests that genetic factors may have roles in influencing brain sexual differentiation._"

"_Affymetrix microarray chips U74A, version 2, which consists of all sequences (9268) in the current mouse UniGene database (Build 120) that have been functionally characterized, as well as 3220 EST clusters, detected **36 genes with expression levels enhanced in female brain (Table 2)**. One of these genes, Inactive **X specific transcript (Xist)**, exhibited the highest fold change (18.5) compared to its expression level in male brains. The other genes enhanced in female brain had fold change values ranging between 1.23 and 2.4 compared to their expression levels in male brain. Affymetrix chips also detected **18 genes with higher expression levels in male brain (Table 3). Two genes, both located on the Y chromosome, DEAD box polypeptide (Dby) and eukaryotic translation initiation factor 2,Y (Eif2s3y), have been detected with fold changes of 10.0 and 8.8, respectively.** Other genes with male brain enhanced expression have fold change values ranging between 1.25 and 2.46._"

See table 2 and 3 for gene lists

----

[Gender-Specific Gene Expression in Post-Mortem Human Brain: Localization to Sex Chromosomes](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3130534/)
- human adult brain microarray data of various regions, notes again on tissue heterogeneity and expression

"Gender differences in brain development and in the prevalence of neuropsychiatric disorders such as depression have been reported. Gender differences in human brain might be related to patterns of gene expression. Microarray technology is one useful method for investigation of gene expression in brain. We investigated gene expression, cell types, and regional expression patterns of differentially expressed sex chromosome genes in brain. We profiled gene expression in male and female dorsolateral prefrontal cortex, anterior cingulate cortex, and cerebellum using the Affymetrix oligonucleotide microarray platform. **Differentially expressed genes between males and females on the Y chromosome (DBY, SMCY, UTY, RPS4Y, and USP9Y) and X chromosome (XIST) were confirmed using real-time PCR measurements.** In situ hybridization confirmed the differential expression of gender-specific genes and neuronal expression of XIST, RPS4Y, SMCY, and UTY in three brain regions examined. The XIST gene, which silences gene expression on regions of the X chromosome, is expressed in a subset of neurons. Since a subset of neurons express gender-specific genes, neural subpopulations may exhibit a subtle sexual dimorphism at the level of differences in gene regulation and function. The distinctive pattern of neuronal expression of XIST, RPS4Y, SMCY, and UTY and other sex chromosome genes in neuronal subpopulations may possibly contribute to gender differences in prevalence noted for some neuropsychiatric disorders. Studies of the protein expression of these sex- chromosome-linked genes in brain tissue are required to address the functional consequences of the observed gene expression differences."

Other papers I had open on my laptop before the tea incident (thank you mobile phone for being sync'd on the browser).

We might be able to get journal ideas here...

A really nice review by the same last author as our study: 
[Reframing sexual differentiation of the brain - 2011](http://www.nature.com.ezproxy.library.ubc.ca/neuro/journal/v14/n6/full/nn.2834.html)
- simple figures of estradiol-induced differentiation and of views of sexual differentiation of the brain
 + old ("chromosomal sex determines gonadal sex, which determines brain sex") and 
 + new ("importance of genetics and environment are incorporated along with the effects of hormones to provide a more nuanced portrayal of the types of variables that cause sex differences") 

[Sexual differentiation of monoaminergic neurons - genetic or epigenetic? - 1991](http://www.sciencedirect.com/science/article/pii/016622369190047X) 
- dunno, 90s

[Epigenetic mechanisms are involved in sexual differentiation of the brain - 2012](http://link.springer.com/article/10.1007/s11154-012-9202-z)
- another kind of epigenetic mark in the brain, useful commentary on ERα 
-  "In the preoptic area (POA), which is important for male sexual behavior, histones associated with the estrogen receptor (ER) α and aromatase (Arom) gene promoters are differentially acetylated between the sexes, and two subtypes of histone deacetylase (HDAC2 and 4) are associated with the same promoters at higher frequencies in males in the early postnatal period. **Since ERα and Arom are essential genes in masculinization of the brain,** these findings suggest that histone deacetylation in the early postnatal period is involved in masculinization of the brain. Previous reports have demonstrated that **ERα gene expression in the POA is higher in females during the developmental and pubertal periods and in adulthood**, indicating that **sexually dimorphic ERα expression that appears in early postnatal development is maintained until adulthood by epigenetic programming**. The ERα promoter is also more sparsely methylated in females, with an inverse correlation with ERα expression. In addition to the hormonal effect, the amount of** maternal care received during postnatal development has a lasting effect on ERα expression mediated by DNA methylation of its promoter**. Taken together, these results suggest that epigenetic mechanisms play a central role in the transduction and maintenance of early hormonal and social cues to organize sexually differentiated brain functions."

[Estrogen Masculinizes Neural Pathways and Sex-Specific Behaviors - 2009 Cell](http://www.sciencedirect.com/science/article/pii/S0092867409009167)

[Sex-specific exons control DNA methyltransferase in mammalian germ cells](http://dev.biologists.org/content/125/5/889.short)

[Sex-Specific Parent-of-Origin Allelic Expression in the Mouse Brain](http://science.sciencemag.org/content/329/5992/682)
- interesting, maybe not directly relevant, on imprinting 
 + "Genomic imprinting results in preferential gene expression from paternally versus maternally inherited chromosomes. We used a genome-wide approach to uncover sex-specific parent-of-origin allelic effects in the adult mouse brain. Our study identified preferential selection of the maternally inherited X chromosome in glutamatergic neurons of the female cortex. Moreover, analysis of the cortex and hypothalamus identified 347 autosomal genes with sex-specific imprinting features. In the hypothalamus, sex-specific imprinted genes were mostly found in females, which suggests parental influence over the hypothalamic function of daughters. We show that **interleukin-18**, a gene linked to diseases with sex-specific prevalence, is subject to complex, regional, and sex-specific parental effects in the brain. Parent-of-origin effects thus provide new avenues for investigation of sexual dimorphism in brain function and disease."
 
**Notes for background research**
===================================================================

Something worth watching for: **Eif2s3** (on X chr). Because you found it in the male/female DEG/DMR analysis (and the literature finds the same) and if it is high in vehicle females and low in females treated with ZEB, great, because this gene is a) not picked up by the original paper, b) very important in x-inactivation.

For my notes on Eif2s3, will look into this more:
http://www.sciencedirect.com/science/article/pii/S1534580704004071
http://www.ncbi.nlm.nih.gov/pubmed/16325480
http://www.nature.com/mp/journal/v14/n6/full/mp20095a.html

Also, in case it is confusing for anyone, there is a male isoform Eif2s3y, and this is our top DEG between males and females (not expressed in females, which make sense, because it's on the Y chr).

Uhh, mind you, it might actually not be different in ZEB females, because it's [a gene that escapes X inactivation](http://onlinelibrary.wiley.com/doi/10.1046/j.1469-1809.2003.00038.x/full), I'll dig into this if anything interesting comes up.

I'm just seeing if anything interesting pops up.

----

**Mef2c** came up in one of my other m/f DEG papers - transcription factor, (high in gExp, hypo in methylation for females for us)

[Tissue-specific expression and regulation of sexually dimorphic genes in mice](http://genome.cshlp.org/content/16/8/995.full)				
[Table_S4-brain sexually dimorphic genes.xls	](http://genome.cshlp.org/content/16/8/995/suppl/DC1)	
Microarray analysis on mice 	

For my notes, cardiac stuff, some m/f flavor 
http://www.sciencedirect.com/science/article/pii/S0022282807000156
http://link.springer.com/article/10.1007/s00109-008-0385-4#/page-1
https://www.researchgate.net/figure/51871257_fig6_Fig-6-Dmp1-Cre-mediated-Mef2c-deficiency-results-in-elevated-cortical-OPG-expression

Hm, seems **important in embryonic development**, note our mice treated with ZEB are postnatal day two...

[MEF2C, a MADS/MEF2-family transcription factor expressed in a laminar distribution in cerebral cortex.](http://www.pnas.org/content/90/4/1546.short)
 + "Specific isoforms of this factor are found exclusively in **brain** and are robustly expressed by neurons in cerebral cortex. In situ hybridization indicates that the factor is **expressed preferentially in certain neuronal layers of cortex and that expression declines during postnatal development.** The unusual pattern of expression in brain suggests that this transcription factor may be important in the development of cortical architecture."

Generic dev
- [BMK1/ERK5 regulates serum‐induced early gene expression through transcription factor MEF2C
](http://emboj.embopress.org/content/16/23/7054.abstract)
- [Requirement of the MADS-box transcription factor MEF2C for vascular development
](http://dev.biologists.org/content/125/22/4565.short)
- [Mef2c is a direct transcriptional target of ISL1 and GATA factors in the anterior heart field during mouse embryonic development
](http://dev.biologists.org/content/131/16/3931.short)

----

**Sesn1** (induced by the p53 tumor suppressor protein and play a role in the cellular response to DNA damage and oxidative stress) 

Came up in one of my other m/f DEG papers, they also saw it up in females.

in humans
[see SUPPLEMENTARY DATA 3 ](http://www.nature.com/ncomms/2013/131122/ncomms3771/full/ncomms3771.html)- List of genes with evidence of sex-biased expression in at least one CNS region (passing FDR 0.01)

No obvious m/f _role_ scanning the literature though.

----

And of course Esr1 is good to see up in females, that's estrogen receptor.

Note for later, on males/females, epigenetics and behaviour, Complicated, will be interesting to see in ZEB females
http://www.karger.com/Article/FullText/324402
http://www.pnas.org/content/97/26/14737
http://press.endocrine.org/doi/abs/10.1210/en.2009-0649
http://press.endocrine.org/doi/abs/10.1210/en.2009-0955

----

[Smarcb1](https://ghr.nlm.nih.gov/gene/SMARCB1) -  component of the SWI/SNF complex, which is an ATP-dependent chromatin-remodeling complex that has roles in cell proliferation and differentiation, in cellular antiviral activities, inhibition of tumor formation, etc.
PARP-3 - dna repair
[Foxj2](http://www.ncbi.nlm.nih.gov/gene/55810) - transcriptional activator

These guys are familiar (mostly from my work - cancer context) haha. Not in my lit f/m DEG tables though
